module_assays/docs/docs.py

```diff
@@ -0,0 +1,19 @@
+import mongoengine as me
+import datetime
+import shortuuid
+
+class Assays(me.DynamicDocument):
+    id = me.StringField(required=True,primary_key=True)
+    inchikey = me.StringField(required=True)
+    source = me.StringField(required=True)
+    url = me.StringField()
+    modified = me.DateTimeField(default=datetime.datetime.utcnow)
```

Comment:
Good catch, its based on when the object is created so there is an issue with it. What you have would pass but the following would fail without modification of the modified field (i.e. calling save does not update it)


```python
a1 = Assays()
a1.save()
old_modified = a1.modified
a1.inchikey = ABCDE
a1.save()
new_modified = a1.modified
assert old_modified != new_modified
```
Will fail when it should pass (because it was updated)

Reply:
i would argue that example is correctly failing. You haven't altered the `modified` field, so why would it change? It's not within the purview of our data to police itself. It's the responsibility of the modifying code to faithfully modify the data

Reply:
What I mean was that I made a logic mistake there and didn't update the modified field in the code so the modification is not working correct. I believe in your example the assert would pass (but we can check at some point, its not super high priority for now) but while my test case is working correct, logically it is wrong because it was in fact modified and modified was not updated